spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarTemplate.java

```diff
@@ -207,21 +206,21 @@ public void setBeanName(String beanName) {
 	private MessageId doSend(@Nullable String topic, @Nullable T message, @Nullable Schema<T> schema,
 			@Nullable Collection<String> encryptionKeys,
 			@Nullable TypedMessageBuilderCustomizer<T> typedMessageBuilderCustomizer,
-			@Nullable ProducerBuilderCustomizer<T> producerCustomizer) throws PulsarClientException {
+			@Nullable ProducerBuilderCustomizer<T> producerCustomizer) {
 		try {
 			return doSendAsync(topic, message, schema, encryptionKeys, typedMessageBuilderCustomizer,
 					producerCustomizer)
 				.get();
 		}
 		catch (Exception ex) {
```

Comment:
I think we should keep the "unwrapping" of the original exception as it handles the async (interrupted + execution) exceptions nice by inspecting the cause. This way we still treat it as we did before but really just remove the checked nature of it. Wdyt?

```suggestion
		catch (Exception ex) {
			if (ex instanceof PulsarException pex) {
				throw pex;
			}
			PulsarClientException pulsarClientException = PulsarClientException.unwrap(ex);
			throw new PulsarException(pulsarClientException);
		}
```

Reply:
I agree and will make the necessary changes